server/sonar-server/src/main/java/org/sonar/server/es/BulkIndexer.java

```diff
@@ -70,32 +78,64 @@
   private final BulkProcessor bulkProcessor;
   private final AtomicLong counter = new AtomicLong(0L);
   private final SizeHandler sizeHandler;
+  private final BulkProcessorListener bulkProcessorListener;
+  @Nullable
+  private final DbClient dbClient;
+  @Nullable
+  private DbSession dbSession;
+  private Collection<EsQueueDto> esQueueDtos;
 
+
+  public BulkIndexer(DbClient dbClient, EsClient client, String indexName, Size size) {
+    this.dbClient = dbClient;
+    this.client = client;
+    this.indexName = indexName;
+    this.sizeHandler = size.createHandler(Runtime2.INSTANCE);
+    this.bulkProcessorListener = new BulkProcessorListener();
+    this.bulkProcessor = BulkProcessor.builder(client.nativeClient(), bulkProcessorListener)
+      .setBackoffPolicy(BackoffPolicy.exponentialBackoff())
+      .setBulkSize(FLUSH_BYTE_SIZE)
+      .setBulkActions(FLUSH_ACTIONS)
+      .setConcurrentRequests(sizeHandler.getConcurrentRequests())
+      .build();
+  }
+
+  @Deprecated
   public BulkIndexer(EsClient client, String indexName, Size size) {
+    this.dbClient = null;
     this.client = client;
     this.indexName = indexName;
     this.sizeHandler = size.createHandler(Runtime2.INSTANCE);
-    this.bulkProcessor = BulkProcessor.builder(client.nativeClient(), new BulkProcessorListener())
+    this.bulkProcessorListener = new BulkProcessorListener();
+    this.bulkProcessor = BulkProcessor.builder(client.nativeClient(), bulkProcessorListener)
       .setBackoffPolicy(BackoffPolicy.exponentialBackoff())
       .setBulkSize(FLUSH_BYTE_SIZE)
       .setBulkActions(FLUSH_ACTIONS)
       .setConcurrentRequests(sizeHandler.getConcurrentRequests())
       .build();
   }
 
-  @Override
+  @Deprecated
   public void start() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)